core/meta/inc/TClass.h

```diff
@@ -131,6 +131,22 @@ friend class ROOT::Internal::TCheckHashRecursiveRemoveConsistency;
                         // in ROOT Meta w/o interpreter information
    };
 
+   class ObjectPtr
+   {
+      void *fPtr = nullptr;
+
+      TVirtualStreamerInfo *fAllocator = nullptr;
+
+   public:
+      ObjectPtr(void *ptr, TVirtualStreamerInfo *allocator = nullptr) : fPtr(ptr), fAllocator(allocator) {}
```

Comment:
Unless you skip the default constructor for a reason, it could simplify slightly the `ObjectPtr{nullptr, nullptr}` expressions to `ObjectPtr()`.